From efcc90745188c352e2750ea614f459d93a6e34f5 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 17 Sep 2015 15:44:31 -0500
Subject: [PATCH 52/97] apply backport patch
 collateral-evolutions/network/0035-skb_no_fcs/igb_skb_no_fcs.patch

---
 backports/drivers/net/ethernet/intel/igb/igb_main.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/backports/drivers/net/ethernet/intel/igb/igb_main.c b/backports/drivers/net/ethernet/intel/igb/igb_main.c
index 1b4ca3f..85715cf 100644
--- a/backports/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/backports/drivers/net/ethernet/intel/igb/igb_main.c
@@ -4795,9 +4795,10 @@ static u32 igb_tx_cmd_type(struct sk_buff *skb, u32 tx_flags)
 	cmd_type |= IGB_SET_FLAG(tx_flags, IGB_TX_FLAGS_TSTAMP,
 				 (E1000_ADVTXD_MAC_TSTAMP));
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,4,0)
 	/* insert frame checksum */
 	cmd_type ^= IGB_SET_FLAG(skb->no_fcs, 1, E1000_ADVTXD_DCMD_IFCS);
-
+#endif /* LINUX_VERSION_CODE >= KERNEL_VERSION(3,4,0) */
 	return cmd_type;
 }
 
-- 
2.5.1

